src/MongoDB.Driver.Legacy/MongoCollection.cs

```diff
@@ -165,8 +165,20 @@ private IEnumerable<BsonDocument> Aggregate(IClientSessionHandle session, Aggreg
                 {
                     case "$out":
                         {
-                            var outputCollectionName = lastStage[0].AsString;
-                            outputCollectionNamespace = new CollectionNamespace(_collectionNamespace.DatabaseNamespace, outputCollectionName);
+                            var outValue = lastStage[0];
```

Comment:
Consider adding a regression test.  I don't see one in for Legacy, but perhaps I missed it.

Reply:
done